Sources/CgRPC/shim/channel.c

```diff
@@ -23,44 +23,34 @@
 #include <stdlib.h>
 #include <string.h>
 
-cgrpc_channel *cgrpc_channel_create(const char *address) {
+cgrpc_channel *cgrpc_channel_create(const char *address,
+                                    grpc_arg *args,
+                                    int number_args) {
   cgrpc_channel *c = (cgrpc_channel *) malloc(sizeof (cgrpc_channel));
+
+  grpc_channel_args *channel_args = gpr_malloc(sizeof(grpc_channel_args));
```

Comment:
Do we need to deallocate this structure after use, or is ownership transferred to the channel? The existing code only allocates this on the stack, which would indicate that ownership is not transferred.